src/align_relative_localization/pod_localizer/src/goal_pub.cpp

```diff
@@ -64,10 +64,60 @@ void goal_publisher::laser_data_cb(const sensor_msgs::LaserScanConstPtr& scan)
 	this->laser_data = *scan;
 }
 
+void goal_publisher::prior_cb(const geometry_msgs::PoseStamped& pose_msg)
+{
+	if(!this->prior_set) {
+		this->prior_set = true;
+		this->transformed_prior = false;
+		this->prior_pose = pose_msg;
+		this->pod_prior_lidar_frame = {pose_msg.pose.position.x, pose_msg.pose.position.y};
+		ROS_INFO("[GOAL_PUB] Prior Received");
+	}
+	
+}
+
+bool goal_publisher::inCircle(float_t x, float_t y) {
+
+	return ((pow(x-pod_prior_lidar_frame.first,2) + pow(y-pod_prior_lidar_frame.second,2)) <= pow(CIRCLE_RADIUS, 2));
+
+}
+
 goal_pub_e goal_publisher::get_legs(void)
 {
 	goal_pub_e status = GOAL_PUB_SUCCESS;
+	int left_limit = 0;
+	if(this->prior_set && !this->transformed_prior) {
+
+		try
+		{
+			this->prior_pose.header.frame_id = "/map";
+			this->listener.waitForTransform("/hokuyo", "/map", ros::Time(0), ros::Duration(1.0));
+			this->listener.transformPose("/hokuyo", this->prior_pose, this->prior_pose);
+			this->pod_prior_lidar_frame = {this->prior_pose.pose.position.x, this->prior_pose.pose.position.y};
+			ROS_INFO("[GOAL_PUB] PRIOR (%f, %f)",this->pod_prior_lidar_frame.first, this->pod_prior_lidar_frame.second );
+			this->transformed_prior = true;
+		}
 
+		catch (tf::TransformException &ex)
+		{
+			ROS_ERROR("%s",ex.what());
+			return status;
+		}
+
+	}
+	// float_t angle_left = 0;
+	// int angle_left_index = 0;
+	if(this->prior_set && this->transformed_prior) {
+		ROS_INFO("[GOAL_PUB] PRIOR (%f, %f)",this->pod_prior_lidar_frame.first, this->pod_prior_lidar_frame.second );
+		/*Not Using Angle Limiting as circle check is sufficiently fast*/
+		// std::pair<float_t, float_t> left_point = {this->pod_prior_lidar_frame.first, this->pod_prior_lidar_frame.second -1};
+		// // std::pair<float_t, float_t> right_point = {this->pod_prior_lidar_frame.first, this->pod_prior_lidar_frame.second + 1};
+		// angle_left = atan2(left_point.first, left_point.second);
+		// angle_left = M_PI - angle_left;
+		// angle_left = (angle_left - laser_data.angle_min )/laser_data.angle_increment;
+		// angle_left_index = std::min(0, int(angle_left));
```

Comment:
I think it's a good idea to leave this in the code since it could be a quick performance improvement if necessary at any point